joomla/joomla-cms
# [5.1] Strengthen Security

Pull Request for Issue # .

### Summary of Changes

- Disable PHP execution inside the `media` folder
- Option to apply an `.htaccess` to each filesystem-local folder (defaults to false for b/c)

### Testing Instructions

- Apply the patch.
- create a file `media/test.php` with contents `<?= phpinfo();`
- try to access the file, ie localhost/media/test.php (you shouldn't get the phpinfo page)
- Goto `/administrator/index.php?option=com_plugins&view=plugins` and edit the filesystem local plugin by enabling the `Strengthen Security` switch
- Create a file `images/test.php` with contents `<?= phpinfo();`
- try to access the file, ie localhost/images/test.php (you shouldn't get the phpinfo page)
- Goto `/administrator/index.php?option=com_plugins&view=plugins` and edit the filesystem local plugin by disabling the `Strengthen Security` switch
- check that the images folder doesn't contain an `.htaccess` file

### Actual result BEFORE applying this Pull Request

PHP execution inside the static folders

### Expected result AFTER applying this Pull Request

PHP execution is prevented inside the static folders

### Link to documentations
Please select:
